### ISSUE TITLE
Question: How about support odin serialize?

### ISSUE THREAD

**Issue by JusamLiu** [alelievr/NodeGraphProcessor#87]
Do we have a plan about support odin serialize,beacause it's attribute is very convenient and simple?
eg: drawing List or custom type.

**Comment by alelievr** [comment]
The requirement for supporting odin serialize attributes and unity built-in field attributes are the same: I need to use SerializeField to display the fields in UIElement.

The problem is that currently I can't use it because the fields are not serialized, it will be the case when the big serialization refactor lands, but I don't really know when it will be ready (I still wait for a fix on Unity side).

So to answer your question, it's planned but waiting for the serialization task to be completed :)

**Comment by wqaetly** [comment]
OK, I have almost completed this task, this is my Fork version from the author library, combined with Odin serialization：[https://github.com/wqaetly/NodeGraphProcessor](https://github.com/wqaetly/NodeGraphProcessor)
This is a related blog I wrote to introduce it：[https://www.lfzxb.top/nodegraphprocesssor-and-odin/](https://www.lfzxb.top/nodegraphprocesssor-and-odin/)
Below are some screenshots
![QQ截图20210404172603](https://user-images.githubusercontent.com/35335061/113505309-85e22c00-9570-11eb-9bd5-3df52badd082.png)
![QQ截图20210404172629](https://user-images.githubusercontent.com/35335061/113505311-87135900-9570-11eb-82a2-ca2b3f1b33e8.png)
![QQ截图20210404172705](https://user-images.githubusercontent.com/35335061/113505314-87abef80-9570-11eb-93b6-bfb8df698bc3.png)
![QQ截图20210404172717](https://user-images.githubusercontent.com/35335061/113505315-88448600-9570-11eb-8675-24f3b64e913a.png)
![QQ截图20210404172725](https://user-images.githubusercontent.com/35335061/113505316-88dd1c80-9570-11eb-897c-395cc4201dbf.png)
![QQ截图20210404172956](https://user-images.githubusercontent.com/35335061/113505318-8975b300-9570-11eb-8c88-39898ade6727.png)

**Comment by LudiKha** [comment]
Fantastic work!

**Comment by LudiKha** [comment]
> The requirement for supporting odin serialize attributes and unity built-in field attributes are the same: I need to use SerializeField to display the fields in UIElement.
> 
> The problem is that currently I can't use it because the fields are not serialized, it will be the case when the big serialization refactor lands, but I don't really know when it will be ready (I still wait for a fix on Unity side).
> 
> So to answer your question, it's planned but waiting for the serialization task to be completed :)

Hi @alelievr , any update on this? :)

**Comment by alelievr** [comment]
The serialization refactor has landed but there is still an issue with the ApplySerializedProperty() function that re-creates all the node data which causes all the nodes to lose their references. So I'm waiting for this issue to be fixed to finish implementing the feature.

**Comment by leissler** [comment]
So the serialization refactor is in the master branch, but it's not stable for use in products yet?

**Comment by alelievr** [comment]
Okay, I managed to finish the implementation of SerializedProperties in the node views so now all the builtin Unity property drawers as well as the CustomPropertyDrawers in odin inspector.

I think it's fairly stable now, I'll make a release soon (the 1.3.0)

Here's an example with UnityEvent builtin drawer:

![image](https://user-images.githubusercontent.com/6877923/124520519-053fff80-dded-11eb-9fc0-5d6e292d5040.png)


And the node code:

```CSharp
using UnityEngine;
using GraphProcessor;
using UnityEngine.Events;

[System.Serializable, NodeMenuItem("Custom/Unity Event Node")]
public class UnityEventNode : BaseNode
{
	[Input(name = "In")]
        public float                input;

	[Output(name = "Out")]
	public float				output;

	public UnityEvent			evt;

	public override string		name => "Unity Event Node";
}
```

**Comment by leissler** [comment]
Awesome!!!

**Comment by LudiKha** [comment]
Fantastic! 🦾

**Comment by ivankuzev** [comment]
Yeah saves soo much extra work. Btw there is still some visual problem with lists here (nodes with lists get very tall)
![image](https://user-images.githubusercontent.com/1208701/124741510-70d8b880-df24-11eb-9075-b89d21258002.png)

**Comment by alelievr** [comment]
Hum, it seems to be an issue on GraphView side, I'll see if I can add a workaround somewhere to avoid that issue

**Comment by alelievr** [comment]
I added this bit of code: https://github.com/alelievr/NodeGraphProcessor/commit/35a0514379d252c019545b41525841eb76626440#diff-88aab6f235f642dfb544df6a26caf202382529e8e056a03c2eaf60934faeb4ceR271

Seems to work pretty well

**Comment by ivankuzev** [comment]
Border fixed...node not fixed. here is an example with your list node( See where my mouse is, I can select the node from there. Preventing me to select anything else. works only when the list is unfolded. )
![image](https://user-images.githubusercontent.com/1208701/125789678-01113065-e1f4-4035-9e77-caafd2b9fa0d.png)

**Comment by alelievr** [comment]
Yeah, I can't do much more than that. I've reported the bug in the bug tracker: https://issuetracker.unity3d.com/product/unity/issues/guid/1351693/

Hopefully, it will be fixed soon as it only happens on the 2021.2 betas.
